tests/system/server.go

```diff
@@ -5,43 +5,102 @@ import (
 	"fmt"
 	"os"
 	"strconv"
+	"strings"
+	"time"
 
 	"go.autokitteh.dev/autokitteh/backend/svc"
 	"go.autokitteh.dev/autokitteh/internal/kittehs"
 	"go.autokitteh.dev/autokitteh/tests/internal/svcproc"
 )
 
-var useProcSvc, _ = strconv.ParseBool(os.Getenv("AK_SYSTEST_USE_PROC_SVC"))
+const (
+	serverHTTPAddrFile = "ak_server_addr.txt"
+	serverReadyTimeout = 5 * time.Second
+)
 
-// Start the AK server, but in a goroutine rather than as a separate
-// subprocess: to support breakpoint debugging, and measure test coverage.
-func startAKServer(ctx context.Context, akPath string) (svc.Service, error) {
+// Start the AK server, as an in-process goroutine rather than a separate
+// subprocess (to support breakpoint debugging, and measure test coverage),
+// unless the environment variable AK_SYSTEST_USE_PROC_SVC is set to "true".
+func startAKServer(ctx context.Context, akPath string) (svc.Service, string, error) {
+	runOpts := svc.RunOptions{Mode: "test"}
 	cfg := kittehs.Must1(svc.LoadConfig("", map[string]any{
-		"db.type":   "sqlite",
-		"db.dsn":    "sqlite:autokitteh.sqlite",
-		"http.addr": ":0",
-	}, ""))
+		"db.type": "sqlite",
+		"db.dsn":  "file:autokitteh.sqlite", // In the test's temporary directory.
 
-	runOpts := svc.RunOptions{Mode: "test"}
+		"http.addr":          ":0",
+		"http.addr_filename": serverHTTPAddrFile, // In the test's temporary directory.
+	}, ""))
 
+	// Instantiate the server, either as a subprocess or in-process.
 	var (
-		service svc.Service
-		err     error
+		server svc.Service
+		err    error
 	)
-
-	if useProcSvc {
-		service, err = svcproc.NewSvcProc(akPath, cfg, runOpts)
+	if subproc, _ := strconv.ParseBool(os.Getenv("AK_SYSTEST_USE_PROC_SVC")); subproc {
+		server, err = svcproc.NewSvcProc(akPath, cfg, runOpts)
 	} else {
-		service, err = svc.New(cfg, runOpts)
+		server, err = svc.New(cfg, runOpts)
+	}
+	if err != nil {
+		return nil, "", fmt.Errorf("new AK server: %w", err)
+	}
+
+	// Start the server instance.
+	if err := server.Start(ctx); err != nil {
+		return nil, "", fmt.Errorf("start AK server: %w", err)
 	}
 
+	// Wait for the server's "/readyz" URL to be available.
+	addr, err := waitForReadiness()
 	if err != nil {
-		return nil, fmt.Errorf("svc.New: %w", err)
+		return nil, "", fmt.Errorf("wait for AK server: %w", err)
 	}
 
-	if err := service.Start(ctx); err != nil {
-		panic(fmt.Errorf("fx app start: %w", err))
+	return server, addr, nil
+}
+
+// Wait for the AK server to be ready, with a timeout,
+// and return its HTTP address for client connections.
+func waitForReadiness() (string, error) {
+	ready := make(chan bool, 1)
+	timer := time.NewTimer(serverReadyTimeout)
+	go queryReadyz(ready)
```

Comment:
maybe to return file you already read. In case of success you are reading it twice